src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -39,19 +39,8 @@ protected override string GetInsertionText(ISymbol symbol, AbstractSyntaxContext
 
         public static string GetInsertionText(ISymbol symbol, AbstractSyntaxContext context)
         {
-            string name;
-
-            if (CommonCompletionUtilities.TryRemoveAttributeSuffix(symbol, context.IsAttributeNameContext, context.GetLanguageService<ISyntaxFactsService>(), out name))
-            {
-                // Cannot escape Attribute name with the suffix removed. Only use the name with
-                // the suffix removed if it does not need to be escaped.
-                if (name.Equals(name.EscapeIdentifier()))
-                {
-                    return name;
-                }
-            }
-
-            return symbol.Name.EscapeIdentifier(isQueryContext: context.IsInQuery);
+            var name = CommonCompletionUtilities.GetAppropriateNameInContext(symbol, context);
```

Comment:
Won't this cause a minimal-display-string calculation for every item in the completion list? I got the impression the current design was because these symbols are largely in the correct context and don't need such processing.